src/GafferScene/Unencapsulate.cpp

```diff
@@ -0,0 +1,287 @@
+		// If there's no capsule, we're not expanding anything, just pass through the source scene
+		if( capsule )
+		{
+			return IECore::NullObject::defaultNullObject();
+		}
+		else
+		{
+			return o;
+		}
+	}
+	return capsule->scene()->object( concatScenePath( capsule->root(), branchPath ) );
+}
+
+bool Unencapsulate::affectsBranchChildNames( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+void Unencapsulate::hashBranchChildNames( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	if( !capsule )
+	{
+		h.append( 0 );
+		return;
+	}
+	if( branchPath.size() == 0 )
+	{
+		h = capsule->scene()->childNamesHash( capsule->root() );
+	}
+	else
+	{
+		h = capsule->scene()->childNamesHash( concatScenePath( capsule->root(), branchPath ) );
+	}
+}
+
+IECore::ConstInternedStringVectorDataPtr Unencapsulate::computeBranchChildNames( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	if( !capsule )
+	{
+		return new InternedStringVectorData();
+	}
+
+	if( branchPath.size() == 0 )
+	{
+		return capsule->scene()->childNames( capsule->root() );
+	}
+	else
+	{
+		return capsule->scene()->childNames( concatScenePath( capsule->root(), branchPath ) );
+	}
+}
+
+bool Unencapsulate::affectsBranchSetNames( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+void Unencapsulate::hashBranchSetNames( const ScenePath &parentPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	h = inPlug()->setNamesPlug()->hash();
+}
+
+IECore::ConstInternedStringVectorDataPtr Unencapsulate::computeBranchSetNames( const ScenePath &parentPath, const Gaffer::Context *context ) const
+{
+	// We have a standard that any seen containing capsules must contain all the sets used in the capsules
+	// in their list of set names, even if those sets are empty until the capsules are expanded
+	return inPlug()->setNamesPlug()->getValue();
+}
+
+bool Unencapsulate::affectsBranchSet( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+void Unencapsulate::hashBranchSet( const ScenePath &parentPath, const IECore::InternedString &setName, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	BranchCreator::hashBranchSet( parentPath, setName, context, h );
+
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	if( !capsule )
+	{
+		h.append( 0 );
```

Comment:
Must call base class first, but could use the `setPlug()->defaultValue()` trick instead.